src/overlays/actors/ovl_Demo_Effect/z_demo_effect.c

```diff
@@ -20,121 +104,2126 @@ const ActorInit Demo_Effect_InitVars = {
     (ActorFunc)DemoEffect_Update,
     NULL,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Effect/func_80970F50.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Effect/func_80970F58.s")
+// Code only matches when this is an array, but the second element isn't used. This variable assures only one jewel will
+// play SFX
+s16 sfxJewelId[] = { 0, 0 };
+
+// The object used by the effectType
+s16 effectTypeObjects[] = {
+    /* 0x00 */ OBJECT_EFC_CRYSTAL_LIGHT,
+    /* 0x01 */ OBJECT_EFC_FIRE_BALL,
+    /* 0x02 */ OBJECT_GAMEPLAY_KEEP,
+    /* 0x03 */ OBJECT_EFC_LGT_SHOWER,
+    /* 0x04 */ OBJECT_GOD_LGT,
+    /* 0x05 */ OBJECT_GOD_LGT,
+    /* 0x06 */ OBJECT_GOD_LGT,
+    /* 0x07 */ OBJECT_LIGHT_RING,
+    /* 0x08 */ OBJECT_TRIFORCE_SPOT,
+    /* 0x09 */ OBJECT_GI_MEDAL,
+    /* 0x0A */ OBJECT_GI_MEDAL,
+    /* 0x0B */ OBJECT_GI_MEDAL,
+    /* 0x0C */ OBJECT_GI_MEDAL,
+    /* 0x0D */ OBJECT_GI_MEDAL,
+    /* 0x0E */ OBJECT_GI_MEDAL,
+    /* 0x0F */ OBJECT_EFC_TW,
+    /* 0x10 */ OBJECT_LIGHT_RING,
+    /* 0x11 */ OBJECT_LIGHT_RING,
+    /* 0x12 */ OBJECT_GAMEPLAY_KEEP,
+    /* 0x13 */ OBJECT_GI_JEWEL,
+    /* 0x14 */ OBJECT_GI_JEWEL,
+    /* 0x15 */ OBJECT_GI_JEWEL,
+    /* 0x16 */ OBJECT_GI_JEWEL,
+    /* 0x17 */ OBJECT_GI_M_ARROW,
+    /* 0x18 */ OBJECT_EFC_TW,
+    /* 0x19 */ OBJECT_EFC_TW
+};
+
+u8 timewarpVertexSizeIndices[] = { 1, 1, 2, 0, 1, 1, 2, 0, 1, 2, 0, 2, 1, 0, 1, 0, 2, 0, 2, 2, 0 };
+
+Color_RGB8 jewelSparkleColors[5][2] = { { { 0xFF, 0xFF, 0xFF }, { 0x64, 0xFF, 0x00 } },
+                                        { { 0xFF, 0xFF, 0xFF }, { 0xC8, 0x00, 0x96 } },
+                                        { { 0xFF, 0xFF, 0xFF }, { 0x00, 0x64, 0xFF } },
+                                        { { 0x00, 0x00, 0x00 }, { 0x00, 0x00, 0x00 } },
+                                        { { 0xDF, 0x00, 0x00 }, { 0x00, 0x00, 0x00 } } };
+
+/**
+ * Sets up the update function.
+ */
+void DemoEffect_SetupUpdate(DemoEffect* this, DemoEffectFunc updateFunc) {
+    this->updateFunc = updateFunc;
+}
+
+/**
+ * Gives a number on the range of 0.0f - 1.0f representing current cutscene action completion percentage.
+ */
+f32 DemoEffect_InterpolateCsFrames(GlobalContext* globalCtx, s32 csActionId) {
+    f32 interpolated = func_8006F93C(globalCtx->csCtx.npcActions[csActionId]->endFrame,
+                                     globalCtx->csCtx.npcActions[csActionId]->startFrame, globalCtx->csCtx.frames);
+    if (interpolated > 1.0f) {
+        interpolated = 1.0f;
+    }
+    return interpolated;
+}
+
+/**
+ * Initializes information for Jewel/Spritual Stone actors.
+ */
+void DemoEffect_InitJewel(GlobalContext* globalCtx, DemoEffect* this) {
+    this->initDrawFunc = &DemoEffect_DrawJewel;
+    if (LINK_IS_CHILD) {
+        this->initUpdateFunc = &DemoEffect_UpdateJewelChild;
+    } else {
+        this->initUpdateFunc = &DemoEffect_UpdateJewelAdult;
```

Comment:
```suggestion
        this->initUpdateFunc = DemoEffect_UpdateJewelChild;
    } else {
        this->initUpdateFunc = DemoEffect_UpdateJewelAdult;
```
Functions are their own pointer, so these aren't needed.

Reply:
Good catch, you're right.